aws/resource_aws_default_vpc.go

```diff
@@ -50,44 +50,46 @@ func resourceAwsDefaultVpcCreate(d *schema.ResourceData, meta interface{}) error
 
 	resp, err := conn.DescribeVpcs(req)
 	if err != nil {
+		d.SetId("vpc-removed")
 		return nil
 	}
 
 	if resp.Vpcs == nil || len(resp.Vpcs) == 0 {
+		d.SetId("vpc-removed")
 		return nil
 	}
-
-	d.SetId(aws.StringValue(resp.Vpcs[0].VpcId))
-
-	vpcID := d.Id()
-	deleteVpcOpts := &ec2.DeleteVpcInput{
-		VpcId: &vpcID,
-	}
-	log.Printf("[INFO] Deleting VPC: %s", d.Id())
-
-	err2 := resource.Retry(5*time.Minute, func() *resource.RetryError {
-		_, err2 := conn.DeleteVpc(deleteVpcOpts)
-		if err2 == nil {
-			return nil
-		}
-
-		if isAWSErr(err2, "InvalidVpcID.NotFound", "") {
-			return nil
-		}
-		if isAWSErr(err2, "DependencyViolation", "") {
-			return resource.RetryableError(err2)
+	if resp.Vpcs != nil || len(resp.Vpcs) > 0 {
```

Comment:
same as above, maybe a && ?  you're safe because on 57 we check for resp.Vpcs == nil, but if that line ended up getting removed then this code isn't going to keep you safe